components/core/src/streaming_archive/reader/Segment.hpp

```diff
@@ -58,7 +58,7 @@ namespace streaming_archive { namespace reader {
 
 #if USE_PASSTHROUGH_COMPRESSION
         streaming_compression::passthrough::Decompressor m_decompressor;
-#else
+#elif USE_ZSTD_COMPRESSION
```

Comment:
do we want an "else error" clause?  so it's more obvious to reader that  USE_PASSTHROUGH_COMPRESSION or USE_ZSTD_COMPRESSION must be true.

Reply:
Good point. Addressed.